modules/generic_worker/manifests/multiuser.pp

```diff
@@ -0,0 +1,107 @@
+    String $livelog_secret,
+    String $worker_group,
+    String $worker_type,
+    String $quarantine_client_id,
+    String $quarantine_access_token,
+    String $bugzilla_api_key,
+    String $user,
+    String $user_homedir,
+    Pattern[/^v\d+\.\d+\.\d+$/] $generic_worker_version,
+    String $generic_worker_sha256,
+    Pattern[/^v\d+\.\d+\.\d+$/] $taskcluster_proxy_version,
+    String $taskcluster_proxy_sha256,
+    Pattern[/^v\d+\.\d+\.\d+$/] $quarantine_worker_version,
+    String $quarantine_worker_sha256,
+    Pattern[/^v\d+\.\d+\.\d+$/] $livelog_version,
+    String $livelog_sha256,
+    String $taskcluster_host = 'taskcluster',
+) {
+
+    include httpd
+
+    class { 'packages::generic_worker':
+        generic_worker_version    => $generic_worker_version,
+        generic_worker_sha256     => $generic_worker_sha256,
+        taskcluster_proxy_version => $taskcluster_proxy_version,
+        taskcluster_proxy_sha256  => $taskcluster_proxy_sha256,
+        quarantine_worker_version => $quarantine_worker_version,
+        quarantine_worker_sha256  => $quarantine_worker_sha256
+        livelog_version           => $livelog_version,
+        livelog_sha256            => $livelog_sha256
+    }
+
+    class { 'generic_worker::control_bug':
+        user_homedir     => $user_homedir,
+        bugzilla_api_key => $bugzilla_api_key,
+    }
+
+    $livelog_certificate = "${user_homedir}/livelog.crt"
+    $livelog_key         = "${user_homedir}/livelog.key"
+    $task_dir            = "${user_homedir}/tasks"
+    $caches_dir          = "${user_homedir}/caches"
+    $downloads_dir       = "${user_homedir}/downloads"
+    $ed25519_signing_key = "${user_homedir}/generic-worker.ed25519.signing.key"
+
+    exec {
+        'create ed25519 signing key':
+            path    => ['/bin', '/sbin', '/usr/local/bin', '/usr/bin'],
+            user    => $user,
+            cwd     => $user_homedir,
+            command => "generic-worker new-ed25519-keypair --file ${ed25519_signing_key}",
+            unless  => "test -f ${ed25519_signing_key}",
+            require => Class['packages::generic_worker'];
+    }
+
+    case $::operatingsystem {
+        'Darwin': {
+
+            $reboot_command = '/usr/bin/sudo /sbin/reboot'
+
+            file {
+                '/Library/LaunchDaemons/net.generic.worker.plist':
+                    ensure  => present,
+                    content => template('generic_worker/generic-worker.plist.erb'),
+                    mode    => '0644',
+                    owner   => $::root_user,
+                    group   => $::root_group;
+
+                '/usr/local/bin/run-generic-worker.sh':
+                    ensure  => present,
+                    content => template('generic_worker/run-generic-worker.sh.erb'),
+                    mode    => '0755',
+                    owner   => $::root_user,
+                    group   => $::root_group;
+
+                '/etc/generic-worker.config':
```

Comment:
You changed the references to the config file to `/etc/generic-worker/config` but this is still putting the config in `/etc/generic-worker.config`